packages/core-v2/src/visual-model/default-visual-model.ts

```diff
@@ -0,0 +1,183 @@
+
+  getIdentifier(): string {
+    return this.model.getIdentifier();
+  }
+
+  getVisualEntity(identifier: EntityIdentifier): VisualEntity | null {
+    return this.entities.get(identifier) ?? null;
+  }
+
+  getVisualEntityForRepresented(represented: RepresentedEntityIdentifier): VisualEntity | null {
+    const identifier = this.sourceToEntity.get(represented);
+    if (identifier === undefined) {
+      return null;
+    }
+    return this.getVisualEntity(identifier);
+  }
+
+  getVisualEntities(): Map<EntityIdentifier, VisualEntity> {
+    return new Map(this.entities);
+  }
+
+  subscribeToChanges(listener: VisualModelListener): UnsubscribeCallback {
+    this.listeners.push(listener);
+    // Return callback to remove the listener.
+    return () => {
+      this.listeners = this.listeners.filter(item => item !== listener);
+    };
+  }
+
+  getModelColor(identifier: string): HexColor | null {
+    return this.models.get(identifier)?.color ?? null;
+  }
+
+  getModelColors(): Map<string, VisualModelData> {
+    return new Map(this.models);
+  }
+
+  addVisualNode(entity: Omit<VisualNode, "id" | "type">): string {
+    return this.model.createEntitySync({
+      ...entity,
+      type: [VISUAL_NODE_TYPE],
+    });
+  }
+
+  addVisualRelationship(entity: Omit<VisualRelationship, "id" | "type">): string {
+    return this.model.createEntitySync({
+      ...entity,
+      type: [VISUAL_RELATIONSHIP_TYPE],
+    });
+  }
+
+  addVisualGroup(entity: Omit<VisualGroup, "id" | "identifier">): string {
+    return this.model.createEntitySync({
+      ...entity,
+      type: [VISUAL_GROUP_TYPE],
+    });
+  }
+
+  updateVisualEntity<T extends VisualEntity>(identifier: EntityIdentifier, entity: Partial<Omit<T, "id" | "type">>): void {
+
+  }
+
+  deleteVisualEntity(identifier: EntityIdentifier): void {
+    this.model.deleteEntitySync(identifier);
+  }
+
+  setModelColor(identifier: string, color: HexColor): void {
+    const entityIdentifier = modelEntityIdentifier(identifier);
+    const entity = this.model.getEntitySync(entityIdentifier);
+    if (entity === null) {
+      // We need to create new entity.
+      this.model.createEntitySync<ModelVisualInformation>({
+        type: [MODEL_VISUAL_TYPE],
+        model: identifier,
+        color
+      });
+    } else {
+      // Change existing entity.
+      this.model.changeEntitySync<ModelVisualInformation>(entityIdentifier, { color });
+    }
+    // Update local state in a synchronous way.
+    // We could wait for change report, but we do not need it at this point in time.
+    const model = this.models.get(identifier);
+    if (model === undefined) {
+      this.models.set(identifier, { color });
+    } else {
+      model.color = color;
+    }
+  }
+
+  deleteModelColor(identifier: string): void {
+    const entityIdentifier = modelEntityIdentifier(identifier);
+    // We know there is no data other then color.
+    this.model.deleteEntitySync(entityIdentifier);
+  }
+
+  getId(): string {
```

Comment:
From the commits in this pull request I felt like that you are trying to use `getIdentifier` instead of `getId`, but maybe here it is specifically `getId` for backwards compatibility.

Reply:
Yes, I would like it to be `getIdentifier` to use consistently full names in the code. But as the visual model is used by the rest of the core-v2 expecting `getId`, there is a need for the implementation of this method.